test/tests/p5.SoundLoop.js

```diff
@@ -0,0 +1,161 @@
+const expect = chai.expect;
+
+describe('p5.SoundLoop', function () {
+  it('can be initialized without any arguments', function () {
+    let sloop = new p5.SoundLoop();
+    expect(sloop._bpm).to.equal(60);
+    expect(sloop.musicalTimeMode).to.be.false;
+    expect(sloop._interval).to.equal(1);
+    expect(sloop._timeSignature).to.equal(4);
+    expect(sloop.isPlaying).to.be.false;
+  });
+  it('can be initialized with all arguments', function () {
+    let sloop = new p5.SoundLoop(() => {}, '8n');
+
+    //bpm
+    sloop.bpm = 85;
+    expect(sloop.bpm).to.equal(85);
+    expect(sloop._bpm).to.equal(85);
+
+    //timeSignature
+    sloop.timeSignature = 6;
+    expect(sloop.timeSignature).to.equal(6);
+    expect(sloop._timeSignature).to.equal(6);
+
+    //interval
+    sloop.interval = 0.1;
+    expect(sloop.interval).to.equal(0.1);
+    expect(sloop._interval).to.equal(0.1);
+    expect(sloop.musicalTimeMode).to.be.false;
+
+    //iterations
+    expect(sloop.iterations).to.equal(0); // cannot set iterations
+  });
+  describe('methods', function () {
+    it('can be started and stopped', function (done) {
+      let count = 0;
+      let sloop = new p5.SoundLoop(() => count++, '2n');
+      let ticks;
+      sloop.bpm = 600;
+      sloop.start();
+      expect(sloop.isPlaying).to.be.true;
+      setTimeout(() => {
+        ticks = sloop.iterations;
+        sloop.stop();
+        expect(sloop.isPlaying).to.be.false;
+        expect(ticks).to.not.equal(0);
+        expect(count).to.be.approximately(ticks, 2);
+        done();
+      }, 1000);
+    });
+    it('can be started and stopped with dealy', function (done) {
```

Comment:
```suggestion
    it('can be started and stopped with delay', function (done) {
```

Reply:
Sorry for the spelling mistakes :}

Reply:
no worries, I'm also prone to typos in strings/comments. It might help to add a spell check plugin for your IDE